library/Amazon/SNS/Verify/Validate.hs

```diff
@@ -146,6 +151,9 @@ handleSubscription =
 data SNSNotificationValidationError
   = BadPem String
   | BadUri String
+  | BadScheme String
+  | NoAuthority String
+  | BadDomain String
```

Comment:
Might it be even better to debug if `BadScheme` also included the value of `validScheme` and `BadDomain` included `validRegPattern` to make sure these values are what you think they are?

Reply:
Yes absolutely. In hindsight, now knowing what my bug actually was, that would've resulted in an error like,

```
Bad domain, aws.blah.com. Expected to match "^localhost$"
```

(Assuming I made a nicer `displayException`.)

Which would've immediately showed me the issue.